### PR TITLE
Add Database handler for users

### PR DESCRIPTION
**Pull request by Osuka42g**
#### What does this PR do?
- Add basic usage handling for CouchDB:
-- save: takes a User and save it into the database. The Amazon id is used as key ID.
-- find: takes the Amazon id and looks for an User with that association.
-- deletes: a document with certain Amazon id. Useful for log out intent.

### CODE REVIEW

File: Sources/User.swift [hunk 1]
```diff
@@ -9,20 +9,20 @@
 import Foundation
 import SwiftyJSON
 
-struct User {
-    let profileIconId: Int
+public struct User {
```
Comment (hientrq) [inline Sources/User.swift:12]: why do you have to make it public?

File: Sources/UsersHandler.swift [hunk 2]
```diff
@@ -7,25 +7,95 @@
...
+            "_id":              user.alexaId as AnyObject,
+            "name":             user.name as AnyObject,
+            "summonerLevel":    user.summonerLevel as AnyObject,
+            "accountId":        user.accountId as AnyObject,
+            "revisionDate":     user.revisionDate as AnyObject,
+            "region":           user.region as AnyObject
+        ]
+        let json = JSON(userDict as AnyObject)
+        
+        database.create(json, callback: { (id: String?, rev: String?, document: JSON?, error: NSError?) in
```
Comment (hientrq) [inline Sources/UsersHandler.swift]: you can use `{ (id, rev, document, error) in  .....  }`. It's cleaner

File: Sources/UsersHandler.swift [hunk 3]
```diff
@@ -7,25 +7,95 @@
...
+            if let error = error {
+                print("Error: \(error.code).")
+                completition(false)
+            } else {
+                completition(true)
```
Comment (hientrq) [inline Sources/UsersHandler.swift:57]: we don't use any of the `id`, `rev`, `document` params above if there is no error?

File: Sources/UsersHandler.swift [hunk 4]
```diff
@@ -7,25 +7,95 @@
...
     
-    func find() {
-    
     
+    /// Retrives an User object of determined Amazon Id.
+    ///
+    /// - Parameter alexaId:
+    /// - Returns: nil if not found
+    public func find(_ alexaId: String, completition: @escaping (_ result: User?) -> ()) {
+        database.retrieve(alexaId) { (document: JSON?, error: NSError?) in
```
Comment (hientrq) [inline Sources/UsersHandler.swift]: Same as above you can omit the parameter type to make it easier to read.

File: Sources/UsersHandler.swift [hunk 5]
```diff
@@ -7,25 +7,95 @@
...
+            if error != nil {
+                print("Error: \(error!.code)")
+                completition(nil)
+            } else {
+                completition(User(with: document!, alexaId: (document?["_id"].stringValue)!))
```
Comment (hientrq) [inline Sources/UsersHandler.swift]: don't force unwrap.

File: Sources/UsersHandler.swift [hunk 6]
```diff
@@ -7,25 +7,95 @@
...
+    ///
+    /// - Parameters:
+    ///   - alexaId: The Alexa Id of the user
+    public func delete(_ alexaId: String, completition: @escaping (_ result: Bool) -> ()) {
+        database.retrieve(alexaId) { (document: JSON?, error: NSError?) in
+            if error != nil {
+                print("Error: \(error!.code)")
+                completition(false)
+            } else {
+                self.database.delete((document?["_id"].stringValue)!, rev: (document?["_rev"].stringValue)!, callback: { (error) in
```
Comment (hientrq) [inline Sources/UsersHandler.swift]: don't force unwrap

File: Sources/UsersHandler.swift (hunk 3 above)
Comment (Osuka42g) [reply Sources/UsersHandler.swift:57]: TODO. Errors handling.

File: Sources/User.swift (hunk 1 above)
Comment (Osuka42g) [reply Sources/User.swift:12]: So we can use it on functions in `UsersHandler`

**Review by hientrq — APPROVED** [review]

**Pull request merged**